tests/integration/helpers.py

```diff
@@ -266,3 +267,169 @@ def get_job_invoked_unit(job: jenkins.jenkinsapi.job.Job, units: typing.List[Uni
+            <triggers/>
+            <disabled>false</disabled>
+        </flow-definition>
+        """
+    )
+
+
+def kubernetes_test_pipeline_script() -> str:
+    """Generate a test pipeline script using the kubernetes plugin.
+
+    Return:
+        The pipeline script
+    """
+    return textwrap.dedent(
+        """
+        podTemplate(yaml: '''
+            apiVersion: v1
+            kind: Pod
+            metadata:
+            labels:
+                some-label: some-label-value
+            spec:
+            containers:
+            - name: httpd
+              image: httpd
+              command:
+              - sleep
+              args:
+              - 99d
+              tty: true
+        ''') {
+        node(POD_LABEL) {
+            stage('Integration Test') {
+            sh '''#!/bin/bash
+                hostname
+            '''
+            }
+        }
+        }"""
+    )
+
+
+def create_secret_file_credentials(
+    unit_web_client: UnitWebClient, kube_config: str
+) -> typing.Optional[str]:
+    """Return the necessary components to create a credentials using the jenkins API.
+
+    Args:
+        unit_web_client: Client for Jenkins's remote access API.
+        kube_config: path to the kube_config file.
+
+    Returns:
+        The tuple of the credentials_id, request payload, file bytestream and headers
+    """
+    url = f"{unit_web_client.web}/credentials/store/system/domain/_/createCredentials"
+    credentials_id = f"kube-config-{secrets.token_hex(4)}"
+    payload = {
+        "json": f"""{{
+            "": "4",
+            "credentials": {{
+                "file": "file0",
+                "id": "{credentials_id}",
+                "description": "Created by API",
+                "stapler-class": "org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl",
+                "$class": "org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl",
+            }},
+        }}"""
+    }
+
+    accept_header = (
+        "text/html,"
+        "application/xhtml+xml,"
+        "application/xml;q=0.9,"
+        "image/avif,image/webp,"
+        "image/apng,"
+        "*/*;q=0.8,"
+        "application/signed-exchange;v=b3;q=0.9'"
+    )
+    headers = {
+        "Accept": accept_header,
+    }
+
+    with open(kube_config, "rb") as kube_config_file:
+        files = [("file0", ("config", kube_config_file, "application/octet-stream"))]
+        logger.info("Creating jenkins credentials, params: %s %s %s", headers, files, payload)
+        res = unit_web_client.client.requester.post_url(
+            url=url, headers=headers, data=payload, files=files
+        )
+        logger.info("Credential created, %s", res.status_code)
+        return credentials_id if res.status_code == 200 else None
+
+
+def create_kubernetes_cloud(
+    unit_web_client: UnitWebClient, kube_config_credentials_id: str
+) -> typing.Optional[str]:
+    """Use the jenkins client to create a new kubernetes cloud.
```

Comment:
how about """Use the Jenkins client to add a Kubernetes cloud for dynamic agent provisioning through pods."""? I think this helps determine what the plugin does :D

Reply:
good idea! I'll update it

Reply:
I updated it.